src/Storage/Repository/PgApplicationRepository.cs

```diff
@@ -0,0 +1,217 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Drawing;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Altinn.Platform.Storage.Configuration;
+using Altinn.Platform.Storage.Interface.Models;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Npgsql;
+using NpgsqlTypes;
+
+namespace Altinn.Platform.Storage.Repository
+{
+    /// <summary>
+    /// Handles applicationMetadata repository. Notice that the all methods should modify the Id attribute of the
+    /// Application, since cosmosDb fails if Id contains slashes '/'.
+    /// </summary>
+    public class PgApplicationRepository : IApplicationRepository, IHostedService
```

Comment:
It's should be unnecessary to make the new repository classes be IHostedService.

Reply:
I can't see how to avoid this as long as we need to be able to invoke both the old and the new repository classes. When the convertion is complete, we can remove the old classes and the IHostedService interface.